src/main/java/htsjdk/samtools/SAMTagUtil.java

```diff
@@ -150,16 +191,10 @@ public static short makeBinaryTag(final String tag) {
      *
      * @param tag Tag name packed as 2 ASCII bytes in a short.
      * @return 2-character String representation of a tag name.
+     * @deprecated use {@link SAMTag#makeStringTag(short)} instead
      */
-    public static String makeStringTag(final short tag) {
-        String ret = stringTags[tag];
-        if (ret == null) {
-            final byte[] stringConversionBuf = new byte[2];
-            stringConversionBuf[0] = (byte)(tag & 0xff);
-            stringConversionBuf[1] = (byte)((tag >> 8) & 0xff);
-            ret = StringUtil.bytesToString(stringConversionBuf);
-            stringTags[tag] = ret;
-        }
-        return ret;
+    @Deprecated
+    public String makeStringTag(final short tag) {
```

Comment:
can be static
```suggestion
    public static String makeStringTag(final short tag) {
```

Unless it needs to be non-static for backward compatibility?

Reply:
Keeping it non-static for backwards compatibility.  I was changing them to be static but I realized that would cause confusing breaks in binary compatibility.

Reply:
This pr is basically me fixing some breaks in compatibility that I had though would be harmless but on further inspection seemd problematic.